src/main/java/com/powsybl/openloadflow/network/VoltageControl.java

```diff
@@ -151,9 +154,29 @@ private static void addMainVoltageControls(List<VoltageControl<?>> voltageContro
 
     /**
      * Find the list of voltage control with merge status as main, connected to a given bus (so including by traversing
-     * non impedant branches).
```

Comment:
I still disagree with the way it is implemented even if I see no regression on main behavior. To be discuss but:
- The method `getHighestPriorityMainVoltageControl()` name is clear, even if it is used only to get the target for the moment: we are looking for the voltage control (as an object) that effectively is responsible of the voltage control, with its controllers and its target value. `Main` refers as the acting voltage control. 
- In main branch, we first look at this acting voltage control, based on hard coded priorities, and we take the associated target V. In two step, because we follow a logic. 
- In the feature you are coded, there is a logic too: the main voltage control (object with controllers) is still the same as  in main branch (right?), but you decide to override its target voltage by a new one (because of bad data, what you want). 
The voltage control management is a hard part of this code, and a clear code is crucial for long term support. Open to discussion of course.

Reply:
After a second thought, I think the issue is maybe just the name of the method `getHighestTargetVPriorityMainVoltageControl`.

Reply:
back to first implementation done in 54e796a7ae033155c5445abc622c1f1d7ff3242f